### DISCUSSION TITLE
How to run autogen fast when doing function calling

### DISCUSSION THREAD

**Discussion by rogerbarretocode** [microsoft/autogen#2268] [Q&A]
### Describe the issue

I have created an agent which calls a custom function which does an api call to YouTube . 

Autogen calls this function 10 times despite me prompting it to call it just 5 times for each search string . 

As a result it takes a lot of time . 

Is there a way to speed up function calling ? 

I want autogen to run as fast as possible for my use case

**Comment by ekzhu** [comment]
Can you share the specific system message and function you are using? I think you can also control each function by checking number of concurrent invocation and "throttling" the excessive invocations by skipping the youtube code inside the functions.
